84-sarcastic-fringehead/technical_specification.md

```diff
@@ -0,0 +1,774 @@
+- Event discovery or publication for auditing purposes
+
+## User Journeys (optional)
+
+All user journeys are already detailed in Lynx Boreal. The operations added in this epic will occur automatically without further action required on the part of either the user or GHGA personnel. 
+
+## API Definitions:
+
+### RESTful/Synchronous:
+
+[UCS HTTP API](#ucs-http-api)  
+[UOS HTTP API](#uos-http-api)  
+[WKVS HTTP API](#wkvs-http-api)  
+[EKSS HTTP API](#ekss-http-api)
+[FIS HTTP API](#fis-http-api)  
+
+### Payload Schemas for Events:
+
+#### ResearchDataUploadBoxState
+```python
+OPEN = "open"
+LOCKED = "locked"
+ARCHIVED = "archived"  # This used to be `CLOSED = "closed"`
+```
+
+#### ResearchDataUploadBox
+```python
+id: UUID4
+state: ResearchDataUploadBoxState
+title: str
+description: str
+last_changed: UTCDatetime
+changed_by: UUID4
+file_upload_box_id: UUID4
+locked: bool
+file_count: int
+size: int
+storage_alias: str
+```
+
+#### FileUploadBox
+```python
+id: UUID4
+locked: bool = False  # if archived is True, then this field must also be True
+archived: bool = False  # New field. If True, this box is permanently locked
+file_count: int
+size: int
+storage_alias: str
+```
+
+#### FileUploadState
+```python
+INIT = "init"  # unchanged, means the file is being uploaded to the inbox
+INBOX = "inbox"  # unchanged, means the file is in the inbox awaiting interrogation
+FAILED = "failed"  # new state, means problem with interrogation, upload, etc.
+INTERROGATED = "interrogated"  # new state, means file interrogation was valid
+ARCHIVED = "archived"  # now means the file is officially in permanent storage
+```
+
+#### FileUpload
+> Outbox event owned by UCS
+```python
+id: UUID4       # Unique identifier for the file upload
+box_id: UUID4   # The ID of the FileUploadBox this FileUpload belongs to
+alias: str      # The submitted alias from the metadata (unique within the box)
+state: FileUploadState = "init"  # The state of the FileUpload
+state_updated: UTCDatetime  # Timestamp of when state was updated
+storage_alias: str  # The storage alias for the inbox bucket
+# Additionally, the following fields exist but are unset until later in the process
+secret_id: str | None
+decrypted_sha256: str | None  # SHA-256 checksum of the entire unencrypted file content
+decrypted_size: int     # The size of the unencrypted file
+part_size: int  # The number of bytes in each file part (last part is likely smaller)
+encrypted_parts_md5: list[str] | None     # Is None until DHFS finishes with file
+encrypted_parts_sha256: list[str] | None  # Is None until DHFS finishes with file
+accession: str | None   # The accession number for the file
+```
+
```

Comment:
Add a note that the corresponding mongodb collection should have a compound unique indexes for `(box_id, alias)`.